### PR TITLE
Bugfix: ACE abort when use %l and the msg len exceed ACE_MAXLOGMSGLEN

### PR DESCRIPTION
**Pull request by louishust**

### CODE REVIEW

**Comment by jwillemsen** [comment]
Please extend ACE/tests/Log_Msg_Test.cpp with a reproducer of the issue you see

**Comment by louishust** [comment]
@jwillemsen extend the test

File: ACE/tests/Log_Msg_Test.cpp [hunk 1]
```diff
@@ -746,6 +746,8 @@ test_format_specs (void)
   ACE_DEBUG ((LM_DEBUG, ACE_TEXT ("%S\n"), SIGINT));
   ACE_DEBUG ((LM_DEBUG, ACE_TEXT ("%S\n"), ACE_NSIG));
   ACE_DEBUG ((LM_DEBUG, ACE_TEXT ("thread id %t\n")));
+  char long_msg[2*ACE_MAXLOGMSGLEN] = {'a'};
+  ACE_DEBUG ((LM_DEBUG, ACE_TEXT ("%l %s\n"), long_msg));
```
Comment (jwillemsen) [inline ACE/tests/Log_Msg_Test.cpp]: Logging a char should be %C not %s

Comment (louishust) [reply ACE/tests/Log_Msg_Test.cpp]: not a char, array

Comment (jwillemsen) [reply ACE/tests/Log_Msg_Test.cpp]: Ok, so should be %c

Comment (louishust) [reply ACE/tests/Log_Msg_Test.cpp]: @jwillemsen fixed

**Comment by jwillemsen** [comment]
Thanks for the test extension and the fix!

**Pull request merged**